enhancements/manifestlist/manifestlist-support.md

```diff
@@ -0,0 +1,467 @@
+
+```go
+	DockerImageManifests []ImageManifest
+```
+
+`ImageManifest` struct would look like the following:
+
+```go
+type ImageManifest struct {
+	Digest       string
+	MediaType    string
+	ManifestSize int64
+	Architecture string
+	OS           string
+	Variant      string
+}
+```
+
+The Image pointed by the `Digest` property contains all regular information (i.e. Config) we can
+see nowadays in a regular Image.
+
+Not all fields in an Image are relevant for a manifest list, therefore other change is to add an
+"omitempty" flag on `DockerImageLayers` property as well as to make `DockerImageMetadata` a
+pointer.
+
+The resulting Image struct would look like:
+
+
+```go
+type Image struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	DockerImageReference string
+	DockerImageMetadata *DockerImage
+	DockerImageMetadataVersion string
+	DockerImageManifest string
+	DockerImageLayers []ImageLayer
+	Signatures []ImageSignature
+	DockerImageSignatures [][]byte
+	DockerImageManifestMediaType string
+	DockerImageConfig string
+	DockerImageManifests []ImageManifest
+}
+```
+
+### Proposed ImageStreamImport changes
+
+Users may start an image import by creating an `ImageStreamImport` object. When using these
+objects the user specifies if it desires to import a full repository or a single tag in a
+repository. An `ImageStreamImport` specification looks like:
+
+```go
+type ImageStreamImportSpec struct {
+	Import bool
+	Repository *RepositoryImportSpec
+	Images []ImageImportSpec
+}
+```
+
+Property `Repository` (when set) indicates that the user intends to import a full repository,
+i.e. multiple tags, while `Images` is a slice indicating that a user wants to import only a set
+of tags from one or more repositories. API reports back the import outcome in the `.status`
+property (some types below were embed to make visualization easier):
+
+```go
+type ImageStreamImportStatus struct {
+	Import *ImageStream
+	Images []ImageImportStatus
+	Repository struct {
+		Status metav1.Status
+		Images []ImageImportStatus
+		AdditionalTags []string
+	}
+}
+```
+In the end, the outcome of a single tag import or a full repository import is represented through
+one or multiple `ImageImportStatus` structs, this struct looks like this:
+
+```go
+type ImageImportStatus struct {
+	Status metav1.Status
+	Image *Image
+	Tag string
+}
+```
+
+`ImageImportStatus` struct has a reference for a single image on its `Image` field. This
+enhancement proposes the addition of another field where child images (manifests within a
+manifes list) would be listed, leaving `ImageImportStatus` struct looking like:
+
+
+```go
+type ImageImportStatus struct {
+	Status metav1.Status
+	Image *Image
+	Tag string
+	Manifests []Image
+}
+```
+
+Here the image that refers to the manifest list would be presented on `Image` property (as it is
+today for regular manifests) while all inner manifests within the manifest list would be listed
+in the `Manifests` slice.
```

Comment:
/cc @spadgett 

fyi, ImageStreamImports are going to show more information about manifest lists, afaik Console uses them